pyannote/core/utils/hierarchy.py

```diff
@@ -43,12 +43,20 @@
 from .distance import cdist
 
 
-def linkage(X, method='single', metric='euclidean'):
+def linkage(X, method='single', metric='euclidean',
+            cannot_link: List[Tuple[int, int]] = None,
+            must_link: List[Tuple[int, int]] = None):
```

Comment:
I'd rather make it more generic and future-proof by simply adding a `**kwargs` placeholder.

Reply:
passing `**kwargs` also to `scipy.cluster.hierarchy.linkage` ?

Reply:
I just updated the branch accordingly.